### PR TITLE
Bug 1189676 - Appending dialog to body if used outside of system

### PR DESCRIPTION
**Pull request by jedireza**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=85763a997bb74520670852837052ab1c1e9acfad

File: shared/js/download/download_ui.js [hunk 1]
```diff
@@ -132,7 +132,13 @@ var DownloadUI = (function() {
     confirm.setAttribute('data-z-index-level', '100');
 
     var screen = document.getElementById('screen');
-    screen.appendChild(confirm);
+
+    if (screen) {
```
Comment (albertopq) [inline shared/js/download/download_ui.js]: nit comment. What about something like:
```javascript
var container = screen || document.body;
container.appendChild(confirm);
```
That way we avoid conditionals, but is up to you.
In any case, we should put the else in the same line than the closing if, just for coherence with the rest of the code base. The code structure is very personal, but that's the agreement we did for gaia in order to maintain the same one.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f0e6b5b5c381325df184d7f8e953b094cfdf1eab

File: shared/js/download/download_ui.js (hunk 1 above)
Comment (jedireza) [inline shared/js/download/download_ui.js]: Good call. I've updated the syntax.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=80399f82d1b3638d54f67ae30e2a0fbbc82ff3d1

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=30d45499c8e22af367cda744f0993055915a5ddb

File: shared/js/download/download_ui.js [hunk 2]
```diff
@@ -121,8 +121,7 @@ var DownloadUI = (function() {
 
   function addConfirm() {
     if (confirm !== null) {
-      confirm.innerHTML = '';
-      return;
+      confirm.parentNode.removeChild(confirm);
```
Comment (jedireza) [inline shared/js/download/download_ui.js]: @albertopq I'm calling this out so you're sure to see it.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0ed05f390b53efc9a26e27b68aa49efa90517d40

**Pull request merged**